### PR TITLE
Create a custom hook used to encapsulate the update functionality

### PR DESCRIPTION
**Pull request by nickgraz**
- Apply to both participant and accelerator deliverable variable update views
- Get rid of superfluous `DeliverableEditVariable` component

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2814?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2814** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2814?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2808** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2808?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3 other dependent PRs ([#2809](https://github.com/terraware/terraware-web/pull/2809) <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2809?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2811](https://github.com/terraware/terraware-web/pull/2811) <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2811?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2813](https://github.com/terraware/terraware-web/pull/2813) <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2813?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by constanzauanini — APPROVED** [review]
Great!

File: src/hooks/useProjectVariablesUpdate/index.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    pendingVariableValues.forEach((pendingValues, variableId) => {
+      const variable = variablesWithValues.find((variableWithValues) => variableWithValues.id === variableId);
+      if (!variable) {
+        // This is impossible if the form is only displaying variables that were initialized within the hook
+        snackbar.toastError(strings.GENERIC_ERROR);
+        return;
+      }
+
+      operations = [
+        ...operations,
```
[the remark this answers is not in the document]
Comment (nickgraz) [reply src/hooks/useProjectVariablesUpdate/index.ts:54]: Good catch! It was being populated in two places previously, but now only one. I'll clean it up

[the remark this answers is not in the document]
Comment (nickgraz) [reply src/hooks/useProjectVariablesUpdate/index.ts:54]: Oh wait... scratch that. Operations is appended to for each `pendingVariableValues` entry.

**Pull request merged**